pkg/ethereum/node.go

```diff
@@ -171,29 +172,47 @@ func (n *Node) Start(ctx context.Context) error {
 		return fmt.Errorf("register RPC handlers: %w", err)
 	}
 
+	n.log.Info().Msg("Starting node services")
+
 	// Register the node itself as the notifiee for network connection events
 	n.host.Network().Notify(n)
 
+	n.startMetadataPublisher()
+
 	// Start the discovery service
-	n.sup.Add(n.disc)
+	go n.runDiscovery(ctx)
 
-	log := log.NewLogger("peer_dialer")
+	// Start the peer dialer service
 	for i := 0; i < 16; i++ {
-		cs := &PeerDialer{
-			host:     n.host,
-			peerChan: n.disc.out,
-			maxPeers: n.cfg.MaxPeerCount,
-			log:      log,
-		}
-		n.sup.Add(cs)
+		go n.runPeerDialer(ctx)
```

Comment:
Do things change when we have just one goroutine? Is there a reason why we need more?

Reply:
My understanding is that more peer dialers give us more capacity to dial multiple discovered peers concurrently. If we have a single peer dialer then it will get clogged as the connect takes some time -
https://github.com/chainbound/valtrack/blob/1cef042250ce4e50f6025927b5b543f105611472/pkg/ethereum/peer_dialer.go#L59-L61

This was inspired by Hermes

Reply:
Ok makes sense!